### PR TITLE
TIG-1189 script to convert genny metrics to cedar format

### PR DESCRIPTION
**Pull request by guoyr**
I had to vendor csvsort because the upstream version didn't support parsing csv into numbers. It's only about 200 lines.

the "total" column needs to be filled in. I'll chat with @rtimmons offline about it.

### CODE REVIEW

File: src/python/genny/csv2.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        self.ts_offset = ts_offset
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        return self._parse_into_intermediate_csv(next(self.raw_reader))
+
+    def _parse_into_intermediate_csv(self, line):
+        for i in range(len(line)):
```
Comment (rtimmons) [inline src/python/genny/csv2.py]: how does this account for the header row?

File: src/python/genny/cedar.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+    cur_out_fh.close()
+
+    return output_files
+
+
+def sort_csv_file(file_name, out_dir):
+    # Sort on Timestamp and Thread.
+    csvsort(pjoin(out_dir, file_name),
+            [0, 1],
```
Comment (rtimmons) [inline src/python/genny/cedar.py]: 0 and 1 seem magical (although I know probably copying from the ticket). Can we use `_Columns` and/or do this based on the header row values?

File: src/python/genny/cedar.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            quoting=csv.QUOTE_NONNUMERIC,
+            has_header=False,
```
Comment (rtimmons) [inline src/python/genny/cedar.py]: is it false though?

File: src/python/genny/cedar.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+            show_progress=True)
```
Comment (rtimmons) [inline src/python/genny/cedar.py]: not sure we want to show progress?

File: src/python/genny/cedar.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+
+def parse_args(argv):
+    parser = argparse.ArgumentParser(
+        description='Convert Genny csv2 perf data to Cedar BSON format',
+    )
+    parser.add_argument('input_file', metavar='input-file', help='path to genny csv2 perf data')
+    parser.add_argument('output_dir', metavar='output_dir',
```
Comment (rtimmons) [inline src/python/genny/cedar.py]: `input-file` (dash) versus `output_dir` (underscore) - can they be consistent?

File: src/python/tests/fixtures/csv2/error_outcome.csv [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+MetricsTime,10014000000
+
+OperationThreadCounts
+MyActor,MyOperation,2
+
+Operations
+timestamp, actor  , thread, operation, duration, outcome, n, ops, errors, size
+12345000000    , MyActor, 0     , MyOperation   , 100     , 0      , 1, 6  , 2     , 40
+12345000000    , MyActor, 0     , MyOperation   , 100     , 2      , 1, 6  , 2     , 40
+12345000000    , MyActor, 0     , MyOperation   , 100     , 0      , 1, 6  , 2     , 40
```
Comment (rtimmons) [inline src/python/tests/fixtures/csv2/error_outcome.csv]: trailing newline

File: src/python/tests/fixtures/csv2/invalid_header_thread_count.csv [hunk 7]
```diff
@@ -0,0 +1,5 @@
+Clocks
+SystemTime,1000128
+MetricsTime,10014
+
+BAD_OPERATION_THREAD_COUNT
```
Comment (rtimmons) [inline src/python/tests/fixtures/csv2/invalid_header_thread_count.csv]: newline

File: src/python/tests/fixtures/csv2/two_op.csv [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+66632088000000, InsertRemove, 4     , Insert   , 40      , 0      , 1, 9  , 0     , 90
+66632088000000, InsertRemove, 0     , Remove   , 100     , 0      , 1, 6  , 2     , 40
+66632403000000, InsertRemove, 0     , Remove   , 410     , 0      , 1, 9  , 3     , 60
+66632661000000, InsertRemove, 0     , Remove   , 80      , 0      , 1, 8  , 6     , 21
+66632088000000, InsertRemove, 4     , Remove   , 280     , 1      , 1, 3  , 2     , 24
+66632316000000, InsertRemove, 1     , Remove   , 123     , 1      , 1, 4  , 2     , 35
+66632088000000, InsertRemove, 3     , Remove   , 130     , 0      , 1, 4  , 1     , 36
+66632245000000, InsertRemove, 2     , Remove   , 139     , 1      , 1, 8  , 7     , 10
+66632088000000, InsertRemove, 2     , Remove   , 110     , 0      , 1, 7  , 2     , 12
+66632088000000, InsertRemove, 1     , Remove   , 20      , 0      , 1, 9  , 0     , 19
```
Comment (rtimmons) [inline src/python/tests/fixtures/csv2/two_op.csv]: newline

File: src/python/genny/csv2.py [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+        # Store a reader that starts at the first line of actual csv data after the headers
+        # have been parsed.
+        self._data_reader = None
+
+        # file handle to raw CSV file; lifecycle is shared with this CSV2 object.
+        self._csv_file = None
+
+        try:
+            # Keep this file open for the duration of this object's lifecycle.
+            self._csv_file = open(csv2_file_name, 'r')
```
[the remark this answers is not in the document]
Comment (guoyr) [reply src/python/genny/csv2.py]: Oh good idea! It didn't occur to me to use a context manager.

The above approach should still be correct but is less idiomatic.

**Pull request merged**